kratos/solving_strategies/schemes/residual_based_bdf_scheme.h

```diff
@@ -0,0 +1,582 @@
+            std::cout << "For higher orders than 2 the time step is assumed to be constant. Sorry for the inconveniences" << std::endl;
+
+        // Adding to the process info
+        Vector bdf_vector(mOrder + 1);
+        for (std::size_t i_order = 0; i_order < mOrder + 1; ++i_order)
+            bdf_vector[i_order] = mBDF[i_order];
+        current_process_info(BDF_COEFFICIENTS) = bdf_vector;
+        
+        KRATOS_CATCH( "" );
+    }
+
+    /**
+     * @brief This function is designed to be called once to perform all the checks needed
+     * on the input provided. 
+     * @details Checks can be "expensive" as the function is designed
+     * to catch user's errors.
+     * @param rModelPart The model of the problem to solve
+     * @return Zero means  all ok
+     */
+
+    int Check(ModelPart& rModelPart) override
+    {
+        KRATOS_TRY;
+
+        const int err = ImplicitBaseType::Check(rModelPart);
+        if(err!=0) return err;
+
+        // Check for minimum value of the buffer index
+        // Verify buffer size
+        KRATOS_ERROR_IF(rModelPart.GetBufferSize() < mOrder + 1) << "Insufficient buffer size. Buffer size should be greater than" << mOrder + 1 << ". Current size is" << rModelPart.GetBufferSize() << std::endl;
+
+        KRATOS_CATCH( "" );
+        
+        return 0;
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+protected:
+
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    struct GeneralVectors
+    {
+        std::vector< Vector > dotun0; /// First derivative
+        std::vector< Vector > dot2un0; /// Second derivative
+    };
+    
+    const std::size_t mOrder; /// The integration order
+    Vector mBDF; /// The BDF coefficients
+    GeneralVectors mVector; /// The structure containing the  derivatives
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+    
+    /**
+     * @brief Performing the update of the derivatives
+     * @param rModelPart The model of the problem to solve
+     * @param rDofSet Set of all primary variables
+     * @param A LHS matrix
+     * @param Dx incremental update of primary variables
+     * @param b RHS Vector
+     */
+    inline void UpdateDerivatives(
+        ModelPart& rModelPart,
+        DofsArrayType& rDofSet,
+        TSystemMatrixType& A,
+        TSystemVectorType& Dx,
+        TSystemVectorType& b
+        )
+    {
+        // Updating time derivatives (nodally for efficiency)
+        const int num_nodes = static_cast<int>( rModelPart.Nodes().size() );
+
+        #pragma omp parallel for
+        for(int i = 0;  i< num_nodes; ++i) {
+            auto it_node = rModelPart.Nodes().begin() + i;
```

Comment:
const?

Reply:
yep

Reply:
Nope, because then the derivatives are updated